crypto/openssl/tpm20_Utils.c

```diff
@@ -202,11 +208,17 @@ int32_t fdo_tpm_get_hmac(const uint8_t *data, size_t data_length, uint8_t *hmac,
 		LOG(LOG_DEBUG, "HMAC created successfully.\n");
 
 	} else {
-
+#if defined(ECDSA256_DA)
 		ret_val = Esys_HMAC_Start(esys_context, hmac_key_handle,
 					  auth_session_handle, ESYS_TR_NONE,
 					  ESYS_TR_NONE, &null_auth,
 					  TPM2_ALG_SHA256, &sequence_handle);
+#else
+		ret_val = Esys_HMAC_Start(esys_context, hmac_key_handle,
+					  auth_session_handle, ESYS_TR_NONE,
```

Comment:
Can we define TPM2_ALG_SHA_TYPE as TPM2_ALG_SHA256 or TPM2_ALG_SHA384 based on "defined(ECDSA256_DA)" at one place( like how u defined PLATFORM_HMAC_SIZE) ? I hope then we can remove the multiple #ifs